### PR TITLE
Refactor test-train-validation split utilities and archive feature generation/one-hot encoding utiils.

### PR DESCRIPTION
**Pull request by khanh111**

### CODE REVIEW

File: create_train_test_split.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+Author: Hamzah Khan
+Date: Jan. 23, 2019
+"""
+
+from collections import defaultdict
+from sklearn.model_selection import train_test_split
+import utils.text as text_utils
+
+from argparse import ArgumentParser
```
Comment (piraka9011) [inline create_train_test_split.py]: nit: package formatting (sort by system/3rd party/local + alphabetic)

File: create_train_test_split.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+# This gives us a 60-20-20 split.
+DEFAULT_RANDOM_SEED = 1
+TRAIN_SPLIT_FRACTION = 0.6
+TEST_SPLIT_FRACTION  = 0.2
+VALIDATION_SPLIT_FRACTION = 0.2
+
+parser = ArgumentParser(description='Tarteel Data Train-Test-Validation Splitter')
+parser.add_argument('-i', '--path_to_quran_json', type=str, default="data/data-uthmani.json",
```
Comment (piraka9011) [inline create_train_test_split.py]: Why not `-p` or `-f` for file?

File: create_train_test_split.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+NUM_SURAHS = 114
+
+def create_train_test_validation_split(ayahs_to_text, train_test_validate_fractions, should_group_identical_text=True, random_seed=DEFAULT_RANDOM_SEED):
+    """
+    Create a train-test-validation split over ayahs with the same text, given the Qur'anic data.
+    Returns a list of lists, each an ayah group, containing the ayah numbers.
+    """
+    train_frac = train_test_validate_fractions[0]
+    test_frac = train_test_validate_fractions[1]
+    validate_frac = train_test_validate_fractions[2]
```
Comment (piraka9011) [inline create_train_test_split.py]: Would prefer a small class `ValidationParams` with properties `train/test/validate_fraction`.
This is fine tho.

File: create_train_test_split.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+    # The fractions should sum to 1.0, or we throw an error.
+    if abs(sum(train_test_validate_fractions) - 1.0) > 1e-6:
+        raise Exception("Train-test-validation fractions do not sum to 1.")
```
Comment (piraka9011) [inline create_train_test_split.py]: +1

File: utils/text.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    If `should_include_bismillah` is True, then bismillah text will be stored under (surah num, 0).
+    """
+    ayah_to_text_dict = {}
+
+    # Extract surah and ayah information and fill in text.
+    for surah_obj in quran_obj[SURAHS_KEY]:
+        surah_num = surah_obj[NUM_KEY]
+
+        for ayah_obj in surah_obj[AYAHS_KEY]:
+            ayah_num                                 = ayah_obj[NUM_KEY]
```
Comment (piraka9011) [inline utils/text.py]: nit: spacing. Let's stick with pep8 :)

File: create_train_test_split.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    help="Path to the Qur'anic text Json file.")
+parser.add_argument('-o', '--output_path', type=str)
+parser.add_argument('-g', '--group_identical_text', action='store_true',
+                    help="""If True, ayahs with identical text will be grouped into one set, not spread across
+                            multiple sets.""")
+parser.add_argument('--train_frac', type=float, default=TRAIN_SPLIT_FRACTION)
+parser.add_argument('--test_frac', type=float, default=TEST_SPLIT_FRACTION)
+parser.add_argument('--validation_frac', type=float, default=VALIDATION_SPLIT_FRACTION)
+parser.add_argument('-s', '--seed', type=int, default=DEFAULT_RANDOM_SEED)
+parser.add_argument('-v', '--verbose', action='store_false')
```
Comment (piraka9011) [inline create_train_test_split.py]: Is this used...?

Comment (khanh111) [reply create_train_test_split.py]: It is now :)

**Pull request merged**